test/explorer/series/struct_test.exs

```diff
@@ -21,6 +21,17 @@ defmodule Explorer.Series.StructTest do
              ]
     end
 
+    test "allow nils" do
+      s = Series.from_list([nil, %{"a" => 1, "b" => 2}, nil])
+      assert s.dtype == {:struct, [{"a", {:s, 64}}, {"b", {:s, 64}}]}
+
+      assert Series.to_list(s) == [
+               %{"a" => nil, "b" => nil},
+               %{"a" => 1, "b" => 2},
+               %{"a" => nil, "b" => nil}
+             ]
```

Comment:
We tried a solution similar to your but I think the issue is that this should be:

```suggestion
      assert Series.to_list(s) == [
               nil,
               %{"a" => 1, "b" => 2},
               nil
             ]
```

And I don't know how to solve this without changing how we are creating structs from the Rust side.

Reply:
Interesting - my understanding is that struct just a cosmetic thing in polars as docs say: 

> Polars Structs are the idiomatic way of working with multiple columns. It is also a free operation i.e. moving columns into Structs does not copy any data!

> Polars expressions always have a Fn(Series) -> Series signature and Struct is thus the data type that allows us to provide multiple columns as input/ouput of an expression. In other words, all expressions have to return a Series object, and Struct allows us to stay consistent with that requirement.

I always thought None in struct list as 'broadcasting' (in my mental model).

```python
> s = pl.Series([None, {"a":1, "b": 2}, {"a": None, "b": None}])
>s.to_list()
[{'a': None, 'b': None}, {'a': 1, 'b': 2}, {'a': None, 'b': None}]
```

If we were to expect Series.to_list([nil, %{...},]) is that possible as we cannot be sure if it was nil or %{"a" => nil, "b" => nil} in the original data ?

Is this just one of those oneway operations ?

Reply:
I see, in that case your patch is correct. I will take a look at mixed keys soon and drop a suggestion.

Reply:
Just to clarify - mixed keys always fails with Table.to_columns() and we can't do much about it. We are trying to support all atom keys or all string keys just like `Ecto.cast`.